crates/sozo/ops/src/migrate/mod.rs

```diff
@@ -82,7 +89,28 @@ where
         rpc_url: String,
         guest: bool,
     ) -> Self {
-        Self { diff, world, txn_config, profile_config, rpc_url, guest }
+        Self { diff, world, txn_config, profile_config, rpc_url, guest, verification_config: None }
+    }
+
+    /// Creates a new migration with verification enabled.
+    pub fn with_verification(
+        diff: WorldDiff,
+        world: WorldContract<A>,
+        txn_config: TxnConfig,
+        profile_config: ProfileConfig,
+        rpc_url: String,
+        guest: bool,
+        verification_config: VerificationConfig,
+    ) -> Self {
+        Self {
+            diff,
+            world,
+            txn_config,
+            profile_config,
+            rpc_url,
+            guest,
+            verification_config: Some(verification_config),
+        }
```

Comment:
Good for minimal changes without impacting the `new`, you could have gone a simpler path following the builder pattern:

```rust
pub fn with_verification(self, verification_config: VerificationConfig) -> Self {
    Self { verification_config: Some(verification_config), ..self }
}
```
Still efficient, no `clone` involved.

If we go with an enum, it will no longer be an option, but that's still relevant since the `new` function would use the variant where verification is disabled.